samples/java-fibonacci-action-pojo/src/main/java/com/example/fibonacci/FibonacciAction.java

```diff
@@ -0,0 +1,42 @@
+/* This code was generated by Akka Serverless tooling.
+ * As long as this file exists it will not be re-generated.
+ * You are free to make changes to this file.
+ */
+
+package com.example.fibonacci;
+
+import akka.NotUsed;
+import akka.japi.Pair;
+import akka.stream.javadsl.Source;
+import com.akkaserverless.javasdk.action.Action;
+import com.akkaserverless.javasdk.action.ActionCreationContext;
+
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Predicate;
+
+public class FibonacciAction extends Action { 
+
+  private boolean isFibonacci(long num) {  // <1>
+    Predicate<Long> isPerfectSquare = (n) -> {
+      long square = (long) Math.sqrt(n);
+      return square*square == n;
+    };
+    return isPerfectSquare.test(5*num*num + 4) || isPerfectSquare.test(5*num*num - 4);
+  }
+
+  private long nextFib(long num) {
+    double result = num * (1 + Math.sqrt(5)) / 2.0;
+    return Math.round(result);
+  }
+
+  public Effect<Number> nextNumber(Number number) {
```

Comment:
will it look for methods returning Effect, or what is the criteria for a method becoming a rpc?

Reply:
For the moment, I'm picking methods with one param and returning Effect. 

We can explore more options later. The goal now is to really get the minimal working.